onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.cc

```diff
@@ -195,37 +220,73 @@ Status T5DecoderSubgraph::CreateInitialFeeds(
                                                        num_beam,
                                                        allocator,
                                                        expanded_hidden_states,
-                                                       true));
+                                                       true,
+                                                       0 /*max_sequence_length*/));
       } else {
         ORT_RETURN_IF_ERROR(expand_buffer_float_func(stream,
                                                      encoder_fetches[j],
                                                      num_beam,
                                                      allocator,
                                                      expanded_hidden_states,
-                                                     true));
+                                                     true,
+                                                     0 /*max_sequence_length*/));
       }
       decoder_feeds.push_back(expanded_hidden_states);
     } else {
+      // past key/value for cross attention does not need to be initialized with max_seq_len since they are static.
+      bool use_max_seq_len = (j - first_past_input_index_) < static_cast<size_t>(2 * num_layers);
+
       OrtValue expanded_cache;
       if (is_output_float16_) {
         ORT_RETURN_IF_ERROR(expand_buffer_float16_func(stream,
                                                        encoder_fetches[j],
                                                        num_beam,
                                                        allocator,
                                                        expanded_cache,
-                                                       false));
+                                                       false,
+                                                       use_max_seq_len ? past_present_share_buffer_max_seq_len : 0));
       } else {
         ORT_RETURN_IF_ERROR(expand_buffer_float_func(stream,
                                                      encoder_fetches[j],
                                                      num_beam,
                                                      allocator,
                                                      expanded_cache,
-                                                     false));
+                                                     false,
+                                                     use_max_seq_len ? past_present_share_buffer_max_seq_len : 0));
       }
       decoder_feeds.push_back(expanded_cache);
     }
   }
 
+  // TODO: This part shares the similar logic with CreateInitialFeeds() in subgraph_gpt.cc. We should refactor it.
```

Comment:
> // TODO: [](http://example.com/codeflow?start=1&length=9)

The function CreateInitialFeeds is a little bit too long. It would great if we can refactor the code instead of adding a TODO here, like moving this logic as a helper function in Subgraph

Reply:
sure